app/main/lib/langid.py

```diff
@@ -68,10 +71,71 @@ def langid(text):
         'language': prediction.language,
         'confidence': prediction.probability
       },
-      'raw': prediction
+      'raw': prediction,
+      'model': 'CLD3',
     }
 
   @staticmethod
   def test():
     cld3.get_language("Some text to check")
     return True
+
+class FastTextLangidProvider:
+# https://fasttext.cc/docs/en/language-identification.html
+  fasttext_model = fasttext.load_model("extra/fasttext_language_id/lid.176.ftz")
+  @staticmethod
+  def langid(text):
+    prediction = list(FastTextLangidProvider.fasttext_model.predict(text))
+    # prediction is a list of tuples, e.g., [('__label__en',), array([0.22517213])]
+
+    language = prediction[0][0].split("__")[-1]
+    prediction[1] = prediction[1].tolist()
+
+    # Use 'fil' for Filipino rather than tl for Tagalog
+    if language=="tl":
+      language = "fil"
+
+    return {
+      'result': {
+        'language': language,
+        'confidence': prediction[1][0]
+      },
+      'raw': prediction,
+      'model': 'FastText',
+    }
+
+  @staticmethod
+  def test():
+    FastTextLangidProvider.fasttext_model.get_language("Some text to check")
+    return True
+
+class HybridLangidProvider:
```

Comment:
Assuming the idea is we run both for a little while to see if the agreement is good enough (disagreement probably only on edge cases) and then disable CLD?